web/src/main/java/edu/softserve/zoo/web/security/AuthTokenFilter.java

```diff
@@ -0,0 +1,62 @@
+package edu.softserve.zoo.web.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+
+/**
+ * Filter which intercepts all request for routes that require authentification.
+ * Checks the validity of the remember-me token and authentificate user for current request.
+ *
+ * @author Ilya Doroshenko
+ */
+@Component
+public class AuthTokenFilter extends UsernamePasswordAuthenticationFilter {
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Autowired
+    private AuthTokenUtils tokenUtils;
+
+    @Value("${token.header}")
+    private String tokenHeader;
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String authToken = httpRequest.getHeader(tokenHeader);
+        String username = tokenUtils.getUsernameFromToken(authToken);
+
+        if (username != null && SecurityContextHolder.getContext().getAuthentication() == null) {
+            UserDetails userDetails = userDetailsService.loadUserByUsername(username);
+
+            if (tokenUtils.validateToken(authToken, userDetails)) {
```

Comment:
Request proceeds as unauthenticated one. Spring security will later allow it or deny it with 401 status (if accessed resources require authentication).